### PR TITLE
Add ListPrioCommandParser and test cases

### LINKED ISSUE #69: Add List by Priority
***
# Feature: List Patients by Priority

- **Purpose:** Allows doctors to view patients based on how serious their case is.
- **Command format:** `listPrio [p/Priority]
- **Example commands:**
    - `listPrio p/low`
- **Acceptable values:**
    - **Priority:** Allowed values are `LOW`, `MEDIUM`,  `HIGH`.
        - Predefined values (LOW, MEDIUM, HIGH) for clear urgency classification.
- **Error message if invalid:** "No patients found with this priority level."
- **Outputs:**
    - **Success:** Displays a list of patients with the specified priority level.
    - **Failure:** "No patients found with this priority level."
 - **Duplicate handling:** Not applicable.

***

### LINKED ISSUE #90: Implement ListPrioCommandParser class

### PR DESCRIPTION
**Pull request by PresleyChew**
fixes #90 , fixes #69

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/ListPrioCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        ListPrioCommand expectedListLowPriorityCommand = new ListPrioCommand(lowPriorityPredicate);
+        assertParseSuccess(parser, " !/lOW", expectedListLowPriorityCommand);
+
+        // for command "listPrio !/NONE"
+        Priority noPriority = new Priority("NONE");
+        PriorityMatchesPredicate noPriorityPredicate = new PriorityMatchesPredicate(noPriority);
+        ListPrioCommand expectedListNoPriorityCommand = new ListPrioCommand(noPriorityPredicate);
+        assertParseSuccess(parser, " !/NONE", expectedListNoPriorityCommand);
+
+    }
```
Comment (yong-zhuo) [inline src/test/java/seedu/address/logic/parser/ListPrioCommandParserTest.java:46]: Test case consider each scenario 👍🏻

File: src/main/java/seedu/address/logic/commands/ListPrioCommand.java [hunk 2]
```diff
@@ -17,8 +17,8 @@ public class ListPrioCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Lists all patients whose Priority "
             + "matches the given Priority level.\n"
-            + "Parameters: Priority\n"
-            + "Example: " + COMMAND_WORD + " HIGH";
+            + "Parameters: !/Priority\n"
+            + "Example: " + COMMAND_WORD + " !/HIGH\n";
```
Comment (TechRaj) [inline src/main/java/seedu/address/logic/commands/ListPrioCommand.java:21]: Nice touch

File: src/main/java/seedu/address/logic/parser/ListPrioCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        String trimmedArgs = userInput.trim();
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(userInput, PREFIX_PRIORITY);
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_PRIORITY);
+        if (!argMultimap.getValue(PREFIX_PRIORITY).isPresent()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListPrioCommand.MESSAGE_USAGE));
+        }
+        String priorityStr = argMultimap.getValue(PREFIX_PRIORITY).orElse("");
+        Priority priority = ParserUtil.parsePriority(priorityStr);
+        PriorityMatchesPredicate predicate = new PriorityMatchesPredicate(priority);
+        return new ListPrioCommand(predicate);
```
Comment (yong-zhuo) [inline src/main/java/seedu/address/logic/parser/ListPrioCommandParser.java:35]: Good check for duplicates before parsing priority

**Review by yong-zhuo — APPROVED** [review]

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 4]
```diff
@@ -101,6 +102,9 @@ public Command parseCommand(String userInput) throws ParseException {
         case AddAllergyCommand.COMMAND_WORD:
             return new AddAllergyCommandParser().parse(arguments);
 
+        case ListPrioCommand.COMMAND_WORD:
+            return new ListPrioCommandParser().parse(arguments);
+
```
Comment (hweinian) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:107]: LGTM

File: src/main/java/seedu/address/logic/parser/ListPrioCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+public class ListPrioCommandParser implements Parser<ListPrioCommand> {
+
+    /**
+     * Parses the given user input string and returns a ListPrioCommand.
+     *
+     * @param userInput The user input string containing the command and its parameters.
+     * @return A ListPrioCommand object containing the parsed priority predicate.
+     * @throws ParseException If the user input does not conform to the expected format or
+     *                        if the required priority prefix is missing.
+     */
```
Comment (hweinian) [inline src/main/java/seedu/address/logic/parser/ListPrioCommandParser.java:23]: Detailed Javadocs

**Review by TechRaj — COMMENTED** [review]
LGTM
thumbs-up-asian

File: src/test/java/seedu/address/logic/parser/ListPrioCommandParserTest.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import seedu.address.logic.commands.ListPrioCommand;
+import seedu.address.model.person.Priority;
+import seedu.address.model.person.PriorityMatchesPredicate;
+
+
+public class ListPrioCommandParserTest {
+    private final ListPrioCommandParser parser = new ListPrioCommandParser();
+
+    @Test
+    public void parse_fieldsValid_success() {
```
Comment (hweinian) [inline src/test/java/seedu/address/logic/parser/ListPrioCommandParserTest.java:21]: extensive and comprehensive test case

**Review by hweinian — APPROVED** [review]
lgtm

**Pull request merged**